topics/migration.md

```diff
@@ -0,0 +1,342 @@
+---
+title: Migration from Redis to Valkey
+linkTitle: Migration
+description: How to migrate from Redis to Valkey
+---
+
+This is a migration guide from Redis Community Edition to Valkey. 
+You will learn how to migrate a standalone Redis server instance and a Redis Cluster. 
+
+This guide provides migration steps for Redis server and Valkey deployed in Docker; however, they should also apply for on-premises deployments. 
+Refer to [install Valkey](installation.md) for installation options.
+
+## Why to migrate to Valkey?
+
+* Valkey is the vendor-neutral and open-source software
+* Enhanced performance with multi-threading and dual-channel replication
+* Improved memory efficiency by using one dictionary per slot in cluster mode and embedding keys in dictionaries. 
+
+### Migration compatibility matrix
+
+You can migrate Redis server to Valkey. 
+The following table provides migration options depending on the Redis version you run:
+
+| Redis | Valkey |
+|-------|--------|
+| 6.x   | 7.2.x  |
+| 7.2.x | 7.2.x  |
+| 7.2.x | 8.0.x  |
+| 7.4   | n/a    |
+
+## Migrate a standalone instance
+
+To migrate a standalone Redis server to Valkey, you have the following options:
+
+* [Physical migration](#physical-migration) by copying the most recent on-disk snapshot from the Redis server to Valkey and starting Valkey server with it
+* [Setting up replication](#repication) between Redis and Valkey 
+* [Migrating specific keys](#migrate-specific-keys) 
+
+The example migration steps are provided for Redis 7.2.5 and Valkey version 7.2.6. 
+
+### Physical migration
+
+This is the easiest and fastest migration method. You make a fresh snapshot of your Redis instance and copy it over to Valkey. Valkey reads the data from the snapshot on startup and restores its contents into memory. The tradeoffs for this method are:
+
+* The downtime to shutdown Redis and wait for Valkey to load the data. 
+* Potential risk of data loss on instances with heavy writes. To prevent it, you must disconnect all active connections before the migration start.
+
+1. Disconnect all active connections to the Redis instance.
+
+2. Make a backup of your Redis instance. 
+
+    a. Connect to Redis and check the number of keys:
+
+        ```
```

Comment:
You should drop the triple-backtick fencing when nesting code blocks in two level deep lists. 

As written, the site renders like this
![Screenshot 2024-09-09 at 1 57 33 PM](https://github.com/user-attachments/assets/a7e9a40f-6df1-486b-843e-64f3c0890f96)

But because you can indents are also interpreted as code blocks the it still works. When I remove 54 and 59 it looks like this rendered on-site:

<img width="854" alt="Screenshot 2024-09-09 at 2 00 41 PM" src="https://github.com/user-attachments/assets/07059e2f-a387-4e92-aeb1-17aa8639a8db">

(You'll also need to remove this for the other alphabetical code blocks.)

Reply:
This was surprising. There's something odd about the indentation here but I believe it should be possible to use triple backticks inside list items too.

If I read https://spec.commonmark.org/0.31.2/#list-items correctly, I believe you should indent the body of list item `2. Make` aligned under "M" in "Make" and align the body of `a. Connect` under the "C" in "Connect", rather than adding 4 spaces for each level.

Example:

````
2. Make a backup of your Redis instance. 

   a. Connect to Redis and check the number of keys:

      ```
      code block
      ```
````

or, if you want four spaces indentation per level, you can add an extra space between "2." and "Make" and between "a." and "Connect".

````
2.  Make a backup of your Redis instance. 

    a.  Connect to Redis and check the number of keys:

        ```
        code block
        ```
````